back/engines/commercial/bulk_import_ideas/spec/services/exporters/idea_html_form_exporter_spec.rb

```diff
@@ -74,4 +74,131 @@
+            src:url('/fonts/WOFF/FontyMcFontFace.woff') format('woff');
+          }
+        @font-face {
+          font-family: 'FontyMcFontFaceItalic';
+          src:url('/fonts/WOFF/FontyMcFontFaceItalic.woff') format('woff');
+        }
+      ")
+      allow(Net::HTTP).to receive(:get).with(URI.parse('https://example.com/fonts/WOFF/FontyMcFontFace.woff')).and_return('NORMAL')
+      allow(Net::HTTP).to receive(:get).with(URI.parse('https://example.com/fonts/WOFF/FontyMcFontFaceItalic.woff')).and_return('ITALIC')
+
+      expect(font_styles).to eq("
+        @font-face {
+            font-family: 'FontyMcFontFace';
+            src:url('data:font/woff;base64,Tk9STUFM') format('woff');
+          }
+        @font-face {
+          font-family: 'FontyMcFontFaceItalic';
+          src:url('data:font/woff;base64,SVRBTElD') format('woff');
+        }
+      ")
+    end
+  end
+
+  describe 'custom_font_url' do
+    it 'returns the existing host in the URL if it is absolute' do
+      custom_font_url = service.send(:custom_font_url, 'https://test.com/fonts.css', 'https://other.com')
+      expect(custom_font_url.to_s).to eq 'https://test.com/fonts.css'
+    end
+
+    it 'returns the URL with the supplied host if it is relative' do
+      custom_font_url = service.send(:custom_font_url, '/fonts.css', 'https://other.com')
+      expect(custom_font_url.to_s).to eq 'https://other.com/fonts.css'
+    end
+  end
+
+  describe 'group_fields' do
+    it 'groups page fields with their first child question, otherwise each question has its own group' do
+      # NOTE: Truncated fields object - only includes what is necessary for the method
+      fields = [
+        { input_type: 'page' },
+        { input_type: 'text_multiloc' },
+        { input_type: 'page' },
+        { input_type: 'html_multiloc' },
+        { input_type: 'page' },
+        { input_type: 'image_files' },
+        { input_type: 'files' },
+        { input_type: 'page' },
+        { input_type: 'page' },
+        { input_type: 'multiline_text' },
+        { input_type: 'text' },
+        { input_type: 'page' }
+      ]
+
+      expect(service.send(:group_fields, fields)).to eq [
+        { input_type: 'page', field_group: { start: true, end: false } },
+        { input_type: 'text_multiloc', field_group: { start: false, end: true } },
+        { input_type: 'page', field_group: { start: true, end: false } },
+        { input_type: 'html_multiloc', field_group: { start: false, end: true } },
+        { input_type: 'page', field_group: { start: true, end: false } },
+        { input_type: 'image_files', field_group: { start: false, end: true } },
+        { input_type: 'files', field_group: { start: true, end: true } },
+        { input_type: 'page', field_group: { start: true, end: true } },
+        { input_type: 'page', field_group: { start: true, end: false } },
+        { input_type: 'multiline_text', field_group: { start: false, end: true } },
```

Comment:
Shouldn't this one have `end: false` as there's still another field on the same page? I don't fully understand the representation.

Reply:
No, this is correct. Each question should be it's own group (start and end both true) unless there is a page before it, in which case the page & first question of the page should be grouped together, so for:

page 1
question 1
question 2
page 2
page 3
question 3
question 4
question 5

The field groups would look like this:

page 1, question 1
question 2
page 2
page 3, question 3
question 4
question 5

Does that make sense? Maybe I'll document the code with this. It might make it clearer.